src/plugins/intel_npu/src/plugin/include/metadata.hpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <stdint.h>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace intel_npu {
+
+constexpr std::string_view MAGIC_BYTES = "OVNPU";
+
+constexpr uint32_t make_version(uint16_t major, uint16_t minor) {
+    return major << 16 | (minor & 0x0000ffff);
+}
+
+constexpr uint32_t METADATA_VERSION_1_0{make_version(1, 0)};
+
+constexpr uint32_t CURRENT_METADATA_VERSION{METADATA_VERSION_1_0};
+
+struct OpenvinoVersion {
+    std::string version;
+    uint32_t size;
```

Comment:
Is there a specific reason for keeping these members public?

Reply:
No, I will set them to private then